gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingSource.java

```diff
@@ -0,0 +1,119 @@
+import gobblin.configuration.WorkUnitState;
+import gobblin.metrics.Tag;
+import gobblin.source.extractor.DataRecordException;
+import gobblin.source.extractor.RecordEnvelope;
+import gobblin.source.extractor.extract.EventBasedSource;
+import gobblin.source.workunit.Extract;
+import gobblin.source.workunit.WorkUnit;
+import gobblin.source.extractor.Extractor;
+import gobblin.configuration.SourceState;
+
+
+/**
+ * A {@link Source} implementation for a simple streaming kafka extractor.
+ *
+ * @author Shrikanth Shankar
+ *
+ */
+public class KafkaSimpleStreamingSource extends EventBasedSource<String, RecordEnvelope<byte[]>> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaSimpleStreamingSource.class);
+
+  public static final String TOPIC_WHITELIST = "gobblin.streaming.kafka.topic";
+  public static final String TOPIC_NAME = "topic.name";
+  public static final String PARTITION_ID = "partition.id";
+  private final Closer closer = Closer.create();
+  public static final Extract.TableType DEFAULT_TABLE_TYPE = Extract.TableType.APPEND_ONLY;
+  public static final String DEFAULT_NAMESPACE_NAME = "KAFKA";
+
+  static public Consumer<String, byte[]> getKafkaConsumer(State state) {
+
+    List<String> brokers = state.getPropAsList(ConfigurationKeys.KAFKA_BROKERS);
+    Properties props = new Properties();
+    props.put("bootstrap.servers", String.join(",", brokers));
+    props.put("group.id", state.getProp(ConfigurationKeys.JOB_NAME_KEY));
+    props.put("enable.auto.commit", "false");
+    props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
+    props.put("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+    Consumer consumer = null;
+    try {
+      consumer = new KafkaConsumer<>(props);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return consumer;
+  }
+
+  @Override
+  public List<WorkUnit> getWorkunits(SourceState state) {
+      Consumer<String, byte[]> consumer = getKafkaConsumer(state);
+      LOG.warn("Consumer is " + consumer);
+      String topic = state.getProp(TOPIC_WHITELIST);
+      List<WorkUnit> workUnits = new ArrayList<WorkUnit>();
+      List<PartitionInfo> topicPartitions;
+      topicPartitions = consumer.partitionsFor(topic);
+      LOG.warn("Partition count is " + topicPartitions.size());
+      for (PartitionInfo topicPartition : topicPartitions){
+        LOG.warn("Partition info is " + topicPartition);
+        Extract extract = this.createExtract(DEFAULT_TABLE_TYPE, DEFAULT_NAMESPACE_NAME, topicPartition.topic());
+        WorkUnit workUnit = WorkUnit.create(extract);
+        workUnit.setProp(TOPIC_NAME, topicPartition.topic());
+        workUnit.setProp(ConfigurationKeys.EXTRACT_TABLE_NAME_KEY, topicPartition.topic());
+        workUnit.setProp(PARTITION_ID, topicPartition.partition());
+        workUnits.add(workUnit);
+      }
+      return workUnits;
+    }
+
+
+  /**
```

Comment:
remove the auto-gen-ed javadoc. 
Replace it with specific javadoc if needed. In this case, probably not.